frontend/src/scenes/events/eventsTableLogic.ts

```diff
@@ -33,64 +38,109 @@ const formatEvents = (events, newEvents, apiUrl) => {
     }
     return eventsFormatted
 }
-// props:
-// - fixedFilters
-// - apiUrl = 'api/event/'
-// - live = false
-export const eventsTableLogic = kea({
+
+export interface EventsTableLogicProps {
+    fixedFilters?: FixedFilters
+    apiUrl?: string // = 'api/event/'
+    live?: boolean // = false
+    key?: string
+}
+
+export interface OnFetchEventsSuccess {
+    events: EventType[]
+    hasNext: boolean
+    isNext: boolean
+}
+
+//from visual inspection of lib/api.js
+//we aren't throwing JS Errors
+export interface ApiError {
+    status?: string
+    statusText?: string
+}
+
+export const eventsTableLogic = kea<eventsTableLogicType<ApiError, EventsTableLogicProps, OnFetchEventsSuccess>>({
+    props: {} as EventsTableLogicProps,
     // Set a unique key based on the fixed filters.
     // This way if we move back/forward between /events and /person/ID, the logic is reloaded.
     key: (props) =>
-        (props.fixedFilters ? JSON.stringify(props.fixedFilters) : 'all') +
-        '-' +
-        (props.apiUrl || 'events') +
-        (props.live ? '-live' : '') +
-        props.key,
-
-    actions: () => ({
-        setProperties: (properties) => {
-            if (Array.isArray(properties) && properties.length === 0) {
-                return { properties: {} }
+        [
+            props.fixedFilters ? JSON.stringify(props.fixedFilters) : 'all',
+            props.apiUrl || 'events',
+            props.live ? 'live' : '',
+            props.key,
+        ].join('-'),
+
+    actions: {
+        setProperties: (properties: Record<string, unknown>[]) => {
+            // there seem to be multiple representations of "empty" properties
+            // the page does not work with some of those representations
+            // this action normalises them
+            if (Array.isArray(properties)) {
+                if (properties.length === 0) {
+                    return { properties: [{}] }
+                } else {
+                    return { properties }
+                }
+            } else {
+                return { properties: [properties] }
             }
-            return { properties }
         },
-        setColumnConfig: (columnConfig) => ({ columnConfig }),
-        setColumnConfigSaving: (saving) => ({ saving }), // Type: boolean
+        setColumnConfig: (columnConfig: string[]) => ({ columnConfig }),
+        setColumnConfigSaving: (saving: boolean) => ({ saving }),
         fetchEvents: (nextParams = null) => ({ nextParams }),
-        fetchEventsSuccess: (events, hasNext = false, isNext = false) => ({ events, hasNext, isNext }),
+        fetchEventsSuccess: (x: OnFetchEventsSuccess) => x,
```

Comment:
I'm not a big fan of short variable names like `x` or `s`. Can you rename to something a bit more explicit? Just `response` if nothing better.

Reply:
Ah, I like an x in the very limited case where it means "the thing you can sort of ignore in this lambda" :)

But, code style trumps individual preference!